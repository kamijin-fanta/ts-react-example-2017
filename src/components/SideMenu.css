.side-menu .active::after {
  content: ' <';
  display: inline;
}
