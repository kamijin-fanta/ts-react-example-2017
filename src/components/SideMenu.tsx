import * as React from 'react';
import { Route, NavLink, Link } from 'react-router-dom';
import { FormSubmitHandler } from 'redux-form';

import { FaHome, FaQuestion } from 'react-icons/lib/fa';

import './SideMenu.css';

export function SideMenu() {
  return (
    <div className="side-menu">
      <h1>SideMenu</h1>
      <ul>
        <li>
          <NavLink to="/" exact={true}>
            <FaHome />
            home
          </NavLink>
        </li>
        <li>
          <NavLink to="/about">
            <FaQuestion />
            about
          </NavLink>
          <ul>
            <li>
              <NavLink to="/about/contact">
                <FaQuestion />
                contact
              </NavLink>
            </li>
          </ul>
        </li>
        <li>
          Path: <Route children={props => props.location.pathname} />
        </li>
      </ul>
    </div>
  );
}
