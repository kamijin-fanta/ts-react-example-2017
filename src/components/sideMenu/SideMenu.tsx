import * as React from 'react';
import { Route, NavLink, Link } from 'react-router-dom';
import { FormSubmitHandler } from 'redux-form';

import { FaHome, FaQuestion, FaListUl } from 'react-icons/lib/fa';
import { Routes } from '../../routes';

import './SideMenu.css';

export function SideMenu() {
  return (
    <div className="side-menu">
      <h1>SideMenu</h1>
      <ul>
        <li>
          <Routes.Hello.NavLink exact={true}>
            <FaHome />
            home
          </Routes.Hello.NavLink>
        </li>
        <li>
          <Routes.About.NavLink>
            <FaQuestion />
            about
          </Routes.About.NavLink>
          <ul>
            <li>
              <Routes.Contact.NavLink>
                <FaQuestion />
                contact
              </Routes.Contact.NavLink>
            </li>
            <li>
              <NavLink to="/about/notfound">notfound</NavLink>
            </li>
          </ul>
        </li>
        <li>
          <Routes.Todo.NavLink>
            <FaListUl />
            todo
          </Routes.Todo.NavLink>
        </li>
        <li>
          <NavLink to="/notfound">notfound</NavLink>
        </li>
        <li>
          Path: <Route children={props => props.location.pathname} />
        </li>
      </ul>
    </div>
  );
}
