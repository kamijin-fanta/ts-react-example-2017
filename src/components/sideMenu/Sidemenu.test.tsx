import * as React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SideMenu } from './SideMenu';

it('side menu snapshot', () => {
  const wrapper = shallow(<SideMenu />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
