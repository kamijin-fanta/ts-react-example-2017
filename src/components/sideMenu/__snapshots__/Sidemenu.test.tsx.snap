// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side menu snapshot 1`] = `
<div
  className="side-menu"
>
  <h1>
    SideMenu
  </h1>
  <ul>
    <li>
      <bound 
        exact={true}
      >
        <FaHome />
        home
      </bound >
    </li>
    <li>
      <bound >
        <FaQuestion />
        about
      </bound >
      <ul>
        <li>
          <bound >
            <FaQuestion />
            contact
          </bound >
        </li>
      </ul>
    </li>
    <li>
      <bound >
        <FaListUl />
        todo
      </bound >
    </li>
    <li>
      Path: 
      <Route />
    </li>
  </ul>
</div>
`;
